include/oneapi/dpl/internal/reduce_by_segment_impl.h

```diff
@@ -1,19 +1,42 @@
-/*
- *  Copyright (c) Intel Corporation
+/*  Copyright (c) Intel Corporation
  *
- *  Copyright 2008-2013 NVIDIA Corporation
+ *  Copyright 2008-2013 NVIDIA Corporation
```

Comment:
That file header is correct?
So we really use NVIDIA code in that file?

Reply:
Even we borrowed some idea, we should comment it in the code with line numbers, I think..
It will help us to make a decision to change file header in the future, when the code will be modified totally, probably.

Reply:
We do not directly use NVIDIA code but enough code was looked at for `wg_segmented_scan` that after discussion with others it was deemed necessary to provide an attribution.

I agree that we should comment in the line numbers as there are only 10 or so lines that apply. I will add this.